coralproject/talk
# Added support for forcing languages

## What does this PR do?

Adds a new `TALK_WHITELISTED_LANGUAGES` configuration parameter:

> This is a **Build Variable** and must be consumed during build. If using the
> [Docker-onbuild](/talk/installation-from-docker/#onbuild)
> image you can specify it with `--build-arg TALK_WHITELISTED_LANGUAGES=en`.
> 
> Specify the comma separated whitelisted languages that you want the Talk
> application to serve. This will override the available set of languages that
> Talk will allow to be served.
> 
> If the [TALK_DEFAULT_LANG](#talk-default-lang) is not included in this list of
> whitelisted languages, then the first whitelisted language will become the
> default language. If this parameter is empty, then all languages supported by
> Talk will be whitelisted. (Default '')

Motivated from the work seen in #1649 by @jerodb!

## How do I test this PR?

1. Set `TALK_WHITELISTED_LANGUAGES=es` in your `.env` file and ensure your browser does not send down the `es` language as supported.
2. Rebuild static assets for Talk (via `yarn build`)
3. Visit the site, notice the language change.
4. Request a password reset, notice that the language of the email is `es`.
